metals/src/main/scala/scala/meta/internal/metals/RamboSymbolSearch.scala

```diff
@@ -0,0 +1,86 @@
+package scala.meta.internal.metals
+
+import scala.meta.pc.SymbolSearch
+import scala.meta.pc.SymbolDocumentation
+import java.{util => ju}
+import org.eclipse.lsp4j.Location
+import scala.meta.pc.SymbolSearchVisitor
+import scala.meta.pc.SymbolSearch.Result
+
+import coursierapi.Dependency
+import coursierapi.Fetch
+import scala.meta.internal.mtags.OnDemandSymbolIndex
+import scala.meta.io.AbsolutePath
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.mtags.Symbol
+import scala.collection.concurrent.TrieMap
+import scala.meta.internal.mtags.Mtags
+
+class RamboSymbolSearch(workspace: AbsolutePath) extends SymbolSearch {
+
+  private val dependencySourceCache =
+    new TrieMap[AbsolutePath, ju.List[String]]()
+
+  private val scalaVersion = BuildInfo.scala212
+
+  private val fetch = Fetch
```

Comment:
Let's move this to the `Embedded` object where we download other jars. We should also update the `DownloadDependencies` class to cache scala-library sources.